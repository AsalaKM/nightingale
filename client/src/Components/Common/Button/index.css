.button {
    background: #555;
    border: none;
    border-radius: 5px;
    outline: none;
    font-family: 'Kodchasan', sans-serif;
    font-style: italic;
    text-align: center;
    color:#fff;
    font-size: 1rem;
    font-weight: 600;
    letter-spacing: 1px;
    cursor: pointer;
    margin: 1rem;
    padding: 0.5rem;
    transition: all .1s;
    min-width: 9rem;
}
.button:hover {
    transform: translateY(-3px);
}

.button:active {
    transform: translateY(3px);

}