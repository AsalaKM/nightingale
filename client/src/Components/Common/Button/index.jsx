import React from "react";
import { Button } from "./index.style";

export default function index(props) {
  const { value, color, onClick } = props;
  return (
    <>
      <Button type="submit" style={{ background: color }} onClick={onClick}>
        {value}
      </Button>
    </>
  );
}
