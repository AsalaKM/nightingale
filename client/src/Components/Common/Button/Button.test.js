import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index.jsx';

describe('Button', () => {
  it('renders correctly', () => {
    const tree = renderer
    .create(<Button value="click me" color="red" onClick={ () => console.log('Hi') }/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  });
})
