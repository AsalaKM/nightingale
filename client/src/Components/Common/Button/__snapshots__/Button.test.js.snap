// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="sc-bdVaJa kigBsQ"
  onClick={[Function]}
  style={
    Object {
      "background": "red",
    }
  }
  type="submit"
>
  click me
</button>
`;
